src/main/java/no/ssb/subsetsservice/Utils.java

```diff
@@ -13,4 +18,13 @@ public static boolean isVersion(String version){
     public static boolean isClean(String str){
         return str.matches("^[a-zA-Z0-9-_]+$");
     }
+
+    public static JsonNode getSelfLinkObject(ObjectMapper mapper, ServletUriComponentsBuilder servletUriComponentsBuilder, JsonNode subset){
+        String subsetVersion = subset.get("version").textValue();
+        ObjectNode hrefNode = mapper.createObjectNode();
+        hrefNode.put("href", servletUriComponentsBuilder.toUriString()+"/"+subsetVersion);
+        ObjectNode self = mapper.createObjectNode();
+        self.set("self", hrefNode);
+        return self;
+    }
```

Comment:
Imho it does not belong here. For me, it looks like it is a subset of responsibility to know how to build its members. It could be practical to gather all subset members related actions in one place.

But on the other hand, any object should have a possibility to build a self-reference. It could be practical to change the self-reference building routine once for all (reusable), then it has to be not dependent on specific calls like `.get("version")` provide all values when calling the method. Let the util be responsible just for the json node structure (Clean code: one task per method).

Reply:
It's not entirely clear to me how I would do this properly, but I have made an issue and will probably tackle it in the near future

Reply:
I see. I believe we will find a better place for this in the code later as application grows. @drageset